test/Marketplace.ts

```diff
@@ -0,0 +1,637 @@
+        ).to.be.revertedWithCustomError(instanceMarket, "NotActiveOffer");
+      });
+      it("fail: attempt to edit the offer by user1", async function () {
+        const { owner, user1, user2, instance721, instance20, instanceMarket } =
+          await loadFixture(standardPrepare);
+        await createOffer(
+          owner,
+          user1,
+          user2,
+          instanceMarket,
+          instance20,
+          instance721,
+          instance20.address,
+          0,
+          100000
+        );
+        await expect(
+          instanceMarket
+            .connect(user1)
+            .editOffer(0, ethers.utils.parseUnits("2", "18"), ZERO_ADDRESS)
+        ).to.be.revertedWithCustomError(instanceMarket, "OnlyOwner");
+      });
+      it("fail: attempt to edit the offer with zero price", async function () {
+        const { owner, user1, user2, instance721, instance20, instanceMarket } =
+          await loadFixture(standardPrepare);
+        await createOffer(
+          owner,
+          user1,
+          user2,
+          instanceMarket,
+          instance20,
+          instance721,
+          instance20.address,
+          0,
+          100000
+        );
+        await expect(
+          instanceMarket.connect(user2).editOffer(0, 0, ZERO_ADDRESS)
+        ).to.be.revertedWithCustomError(instanceMarket, "OnlyPositivePrice");
+      });
+      it("fail: attempt to edit the offer with unsupported sale token", async function () {
+        const { owner, user1, user2, instance721, instance20, instanceMarket } =
+          await loadFixture(standardPrepare);
+        await createOffer(
+          owner,
+          user1,
+          user2,
+          instanceMarket,
+          instance20,
+          instance721,
+          instance20.address,
+          0,
+          100000
+        );
+        await expect(
+          instanceMarket
+            .connect(user2)
+            .editOffer(0, ethers.utils.parseUnits("2", "18"), instance721.address)
+        ).to.be.revertedWithCustomError(instanceMarket, "UnsupportedSaleToken");
+      });
+    });
+    describe("Accept offer", function () {
+      it("success: user1 accept offer with token", async function () {
+        const { owner, user1, user2, instance721, instance20, instanceMarket } =
+          await loadFixture(standardPrepare);
+
+        await createOffer(
+          owner,
+          user1,
+          user2,
+          instanceMarket,
+          instance20,
+          instance721,
+          instance20.address,
+          1,
+          1000000
+        );
+
+        await instance20.connect(user1).approve(instanceMarket.address, 1000000);
+        expect(await instance721.ownerOf(1)).to.eq(user2.address);
+        const tx = await instanceMarket.connect(user1).acceptOffer(0);
+
+        await expect(tx).to.changeTokenBalances(
+          instance20,
+          [user2, instanceMarket, user1],
+          [900000, 100000, -1000000]
+        );
```

Comment:
```suggestion
        await expect(tx).to.changeTokenBalances(
          instance20,
          [user2, instanceMarket, user1],
          [900000, 100000, -1000000]
        );
```

Лучше использовать переменные, чтобы было понятно что за цифры и можно было поменять в одном месте и всё работало